File: 04주차/YH/MinStack.py
class MinStack:

    def __init__(self):
        self.__stack=[]

    def push(self, val: int) -> None:
        self.__stack.append(val)

    def pop(self) -> None:
        if self.__stack:
            self.__stack.pop()

    def top(self) -> int:
        if self.__stack:
            return self.__stack[-1]

    def getMin(self) -> int:
        return min(self.__stack)


# Your MinStack object will be instantiated and called as such:
# obj = MinStack()
# obj.push(val)
# obj.pop()
# param_3 = obj.top()
# param_4 = obj.getMin()

File: 04주차/NH/Min Stack.js
function MinStack() {
    this.stack=[];
  };
  
  MinStack.prototype.push = function(val) {
    this.stack.push(val);
  };
  
  MinStack.prototype.pop = function() {
      return this.stack.pop();
  };
  
  
  MinStack.prototype.top = function() {
      return this.stack[this.stack.length-1];
  };
  
  
  MinStack.prototype.getMin = function() {
      let min=this.stack[0];
      for(let num of this.stack){
          if(num<min)min=num;
      }
      return min;
  };
  